src/bin/cobalt/new.rs

```diff
@@ -443,24 +443,27 @@ fn move_from_drafts_to_posts(
     let mut posts = config.posts.clone();
     posts.include_drafts = true;
     let posts = posts.build()?;
-    let drafts = posts.drafts.clone();
 
     if posts
         .drafts
+        .as_ref()
         .map(|d| d.includes_file(&file))
         .unwrap_or_default()
     {
-        let drafts = drafts.expect("Should have drafts in this if body");
+        let drafts = posts.drafts.expect("Should have drafts in this if body");
         let target = config.source.join("posts").join(
             file.strip_prefix(drafts.subtree())
                 .expect("Should have subtree as prefix"),
         );
         trace!(
             "post is in `drafts_dir` moving it to `posts` directory: {}",
-            target.to_string_lossy()
+            target.display()
         );
-        cobalt_model::files::copy_file(&file, &target)?;
-        fs::remove_file(file)?;
+        if let Some(parent) = target.parent() {
+            fs::create_dir_all(parent)
+                .map_err(|e| format!("Could not create {:?}: {}", parent, e))?;
+        }
+        fs::rename(file, target.clone())?;
```

Comment:
Why the `clone`?

Reply:
I need to stop calling clone to resolve move issues, changed to `&target`